core/src/main/java/org/elasticsearch/discovery/zen/ping/unicast/UnicastZenPing.java

```diff
@@ -565,7 +565,8 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     private PingResponse createPingResponse(DiscoveryNodes discoNodes) {
-        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName, contextProvider.nodeHasJoinedClusterOnce());
+        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName,
```

Comment:
I thought we were settled on the fact that we want only actual getters in our codebase, as that is a common convention. I don't care too much about which one we want to keep, I just want one to go away right now!